python/src/otel/otel_sdk/otel-instrument

```diff
@@ -1,44 +1,160 @@
+    sys.path.append(LAMBDA_LAYER_PKGS_DIR)
+
+from opentelemetry.environment_variables import (
+    OTEL_PROPAGATORS,
+    OTEL_TRACES_EXPORTER,
+    OTEL_PYTHON_DISABLED_INSTRUMENTATIONS,
+)
+from opentelemetry.sdk.environment_variables import OTEL_RESOURCE_ATTRIBUTES
+
+# - Set the default Trace Exporter
+
+environ.setdefault(OTEL_TRACES_EXPORTER, "otlp_proto_grpc_span")
+
+# - Set the service name
+
+service_name_resource_attribute = "service.name"
+if environ.get(OTEL_RESOURCE_ATTRIBUTES) is None:
+    environ[OTEL_RESOURCE_ATTRIBUTES] = "%s=%s" % (
+        service_name_resource_attribute,
+        environ.get(AWS_LAMBDA_FUNCTION_NAME),
     )
-elif "service.name=" not in environ.get("OTEL_RESOURCE_ATTRIBUTES"):
-    environ["OTEL_RESOURCE_ATTRIBUTES"] = "service.name=%s,%s" % (
-        environ.get("AWS_LAMBDA_FUNCTION_NAME"),
-        environ.get("OTEL_RESOURCE_ATTRIBUTES"),
+elif service_name_resource_attribute not in environ.get(
+    OTEL_RESOURCE_ATTRIBUTES
+):
+    environ[OTEL_RESOURCE_ATTRIBUTES] = "%s=%s,%s" % (
+        service_name_resource_attribute,
+        environ.get(AWS_LAMBDA_FUNCTION_NAME),
+        environ.get(OTEL_RESOURCE_ATTRIBUTES),
     )
 
-# TODO: Remove if sdk support resource detector env variable configuration.
+# - Set the Resource Detectors (Resource Attributes)
+#
+#   TODO: waiting on OTel Python support for configuring Resource Detectors from
+#   an environment variable. Replace the bottom code with the following when
+#   this is possible.
+#
+#   environ["OTEL_RESOURCE_DETECTORS"] = "aws_lambda"
+#
 lambda_resource_attributes = (
     "cloud.region=%s,cloud.provider=aws,faas.name=%s,faas.version=%s"
     % (
         environ.get("AWS_REGION"),
-        environ.get("AWS_LAMBDA_FUNCTION_NAME"),
+        environ.get(AWS_LAMBDA_FUNCTION_NAME),
         environ.get("AWS_LAMBDA_FUNCTION_VERSION"),
     )
 )
-environ["OTEL_RESOURCE_ATTRIBUTES"] = "%s,%s" % (
+environ[OTEL_RESOURCE_ATTRIBUTES] = "%s,%s" % (
     lambda_resource_attributes,
-    environ.get("OTEL_RESOURCE_ATTRIBUTES"),
+    environ.get(OTEL_RESOURCE_ATTRIBUTES),
 )
 
-# start the runtime with the extra options
-system(" ".join(args))
+# - Set the default propagators
+
+environ.setdefault(OTEL_PROPAGATORS, "tracecontext,b3,xray")
+
+# - Disable Lambda instrumentation because we will instrument when we know it
+#   patch in a way that is visible to the calling AWS Lambda `boostrap.py` file.
+
+environ[OTEL_PYTHON_DISABLED_INSTRUMENTATIONS] = "aws_lambda"
+
+# - Use a wrapper because instrumentations don't last across parent -> child
+#   processes (i.e. using `excel`) if the child process imports packages
+#   using `getattr(module_name, function_name)`. The child will just import an
+#   uninstrumented package if we don't have `otel_wrapper.py`.
+
+environ["ORIG_HANDLER"] = environ.get("_HANDLER")
+environ["_HANDLER"] = "otel_wrapper.lambda_handler"
+
+# - Call the upstream auto instrumentation script
+
```

Comment:
Strongly recommend rewriting this file into a shell script. It solidifies two facts

1) It's supposed to only do very simple things like setting environment variables, no business logic can happen in here
2) There is no parent python process in our lambda handler, the fact that this file is python I feel can confuse a reader into wondering why we don't load instrumentation here (I think you tried this in a PR once :P). A shell script makes obvious this isn't the app, it's just configuring the environment

Reply:
There's been a lot of talk at the python SIG about moving _all_ bash scripts to python. Python does scripts really well! 

> 1. It's supposed to only do very simple things like setting environment variables, no business logic can happen in here

One huge benefit is being able to import other python packages to GET environment variable constants which like you said is the point of this file. Then we can be sure updates get picked up. (Although they are spec defined things in the instrumentation section are still changing).

All in all a python script should be easier to maintain by a python script.

> 2. There is no parent python process in our lambda handler, the fact that this file is python I feel can confuse a reader into wondering why we don't load instrumentation here (I think you tried this in a PR once :P). A shell script makes obvious this isn't the app, it's just configuring the environment

Lol yes I did but I learned a lot about the Lambda instrumentation in #152  🥲 I added an explicit comment at the top description. I feel like that was what the file was missing and with this it shouldn't confuse future readers.